envoy/network/address.h

```diff
@@ -50,6 +55,11 @@ class Ipv6 {
    * @return true if address is Ipv6 and Ipv4 compatibility is disabled, false otherwise
    */
   virtual bool v6only() const PURE;
+
+  /**
+   * @return Ipv4 address from Ipv4-compatible Ipv6 address.
```

Comment:
Presumably this fails if the v6 address is not in the v4 compatible range? Can you make that more clear in the comments?

Reply:
Yes, will update